src/node/snapshotter.h

```diff
@@ -173,32 +191,47 @@ namespace ccf
 
       if (idx - last_snapshot_idx >= snapshot_tx_interval)
       {
-        auto msg = std::make_unique<threading::Tmsg<SnapshotMsg>>(&snapshot_cb);
-        msg->data.self = shared_from_this();
-        msg->data.snapshot = network.tables->snapshot(idx);
-
         last_snapshot_idx = idx;
 
-        static uint32_t generation_count = 0;
-        threading::ThreadMessaging::thread_messaging.add_task(
-          threading::ThreadMessaging::get_execution_thread(generation_count++),
-          std::move(msg));
+        // Snapshots are only generated on primary node
```

Comment:
Either we pass in is_primary, and that combined with the flag decides what happens and the comment is fair. Or we decide that we abstract away the consensus and we pass `should_emit`, and this comment should not mention anything to do with the primary.